lib/ansible/plugins/inventory/__init__.py

```diff
@@ -219,7 +219,8 @@ def _read_config_data(self, path):
         if not config:
             # no data
             raise AnsibleParserError("%s is empty" % (to_native(path)))
-        elif config.get('plugin') != self.NAME:
+        # If there's a '.' in the plugin name, only fail if there are also no matching collections
+        elif config.get('plugin') != self.NAME and config.get('plugin').split('.')[-1] != self.NAME:
```

Comment:
Not sure of the best way to do this. The alternative would be for NAME in the inventory module to be required to be the full collection name. @bcoca thoughts?

Reply:
you should not need to do this, plugin loader should take care of collection loading itself #56194 should fix the issue of finding 'adjacent' collections for 'non ansible-playbook' , like ansible-inventory

Reply:
just looked at tests, if this is also an issue with playbook, its a bug in loader, the plugins themselves should not need to be 'collection aware'